src/app/GitCommands/CommitMessageManager.cs

```diff
@@ -191,9 +191,11 @@ internal static string FormatCommitMessage(string commitMessage, bool usingCommi
             int lineNumber = 1;
             foreach (string line in commitMessage.LazySplit('\n'))
             {
-                // When a committemplate is used, skip comments and do not count them as line.
-                // otherwise: "#" is probably not used for comment but for issue number
-                if (usingCommitTemplate && line.StartsWith("#"))
+                // When a commit template is used, skip comments and do not count them as line.
+                // Note: In commit templates, "# " is used for comments and "#123" could be used for the issue number.
+                // Redefining the comment symbol is not supported now
+
+                if (usingCommitTemplate && line.StartsWith("# "))
```

Comment:
Pseudocode for hardcoding the desired behavior, without GE setting and minimal user impact

```suggestion
                if (usingCommitTemplate 
                  && ((GitSettingIsNotNull && line.StartsWith(GitSetting))
                    || (GitSettingIsNull && line.StartsWith("#") && (line.Length <= 1 || line[1].IsDigit())))
```